### DISCUSSION TITLE
Developer Question - missing file in github?

### DISCUSSION THREAD

**Discussion by roundface22** [springbok/MLM2PRO-GSPro-Connector#130] [Q&A] [answered]
Hello Developers (including @wonder99 and/or @springbok),

Is src.bluetooth.mlm2pro_secret missing from github, or is it an old reference?

I have a software dev background and I have a SquareGolf Launch monitor on order.  I was looking to see if I could correct some of the phantom putt issues with OCR on the unit.  I've cloned the repository and there seems to be a file missing:  
**mlm2pro_device.py** makes reference to **src.bluetooth.mlm2pro_secret**, which is not included in github.

I have commented out the reference, and the project seems to build and run okay.  I haven't tried anything with MLM2PRO since I don't own one, but, I'm going to assume it's broken.

Otherwise, cloning and running the repo was a breeze.  Cheers gentlemen!

**Comment by wonder99** [comment, marked as answer]
Hi, ya, it's missing, on purpose. Also true that you don't need it, as you discovered.

**Reply by roundface22** [reply]
> Hi, ya, it's missing, on purpose. Also true that you don't need it, as you discovered.

Thanks for the response.
I kinda suspected that was the case, after seeing the methods that were being referenced in there.

**Comment by GaryTheChemist** [comment]
Hey guys! @wonder99 @roundface22 @springbok , sorry im kind of new at this stuff and i wanted to tinker around with the balldata file to come up with some estimated added metrics (I know they wont be accurate but I wanted to give it a try). That said, I have had difficulty in cloning due to the same mentioned issue above. If i ignore, it works but my mlm2pro connects but doesnt authenticate which im guessing is due to the missing secret file? Is there a workaround where i can still use the MLM2pro BT connection?

**Comment by andersdui22** [comment]
Hi @wonder99 @springbok 

I’m developing an open-source MLM2PRO connector. I’ve reached the same point regarding MLM2PROSecret as discussed before.

I saw that you previously offered to share it privately by email. Would you be willing to do the same for my project?

My email is: andersdui22@gmail.com

Thanks for your help!

**Comment by springbok** [comment]
Yes I can send it to you. Are you using my code as the basis for your
connector? If you do, the license requires all source code to be open
source and published. Just mentioning it as I've had people use my code and
then keeping their code private which is a clear violation of the license
agreement.

**Comment by andersdui22** [comment]
Hi Etienne,
Thank you — I completely understand your concern regarding the licence.
This is an independently developed project. I have done my own implementation and testing with my own MLM2PRO, while using your public project as a valuable reference for understanding and verifying parts of the protocol and authorization flow.
I also want to make sure you receive proper credit for your work, so I will include an acknowledgement and link to your project.
If you send me the MLM2PROSecret privately, I will treat it strictly as a credential. It will not be published, committed to a repository, or written to logs.
Thank you again for being willing to help. I really appreciate it.
Best regards,
AndersDen 7. sep. 2026 kl. 02.04 skrev Etienne van Tonder ***@***.***>:﻿
Yes I can send it to you. Are you using my code as the basis for your

connector? If you do, the license requires all source code to be open

source and published. Just mentioning it as I've had people use my code and

then keeping their code private which is a clear violation of the license

agreement.

**Reply by springbok** [reply]
> My email is: ***@***.***

no worries I've sent you the secret
